src/lib.rs

```diff
@@ -311,6 +833,189 @@ impl Session {
+    /// Constructs a `AddrType` from a string
+    pub fn try_addrtype(&self) -> Result<AddrType, ParseEnumError> {
+        AddrType::from_str(self.addrtype.as_str())
+    }
+
+    /// Converts a `AddrType` to string and sets it to the `addrtype`
+    pub fn set_from_addrtype(&mut self, addrtype: AddrType) {
+        self.addrtype = addrtype.to_string();
+    }
+
+    /// Constructs a `NetType` from a string
+    pub fn try_nettype(&self) -> Result<NetType, ParseEnumError> {
+        NetType::from_str(self.nettype.as_str())
+    }
+
+    /// Converts a `NetType` to string and sets it to the `nettype`
+    pub fn set_from_nettype(&mut self, nettype: NetType) {
+        self.nettype = nettype.to_string();
+    }
+
+    /// Constructs a `IpAddr` from a string
+    pub fn try_connection_address(&self) -> Result<IpAddr, AddrParseError> {
+        self.connection_address.parse()
+    }
+
+    /// Converts a `IpAddr` to string and sets it to the `connection_address`
+    pub fn set_connection_address(&mut self, connection_address: IpAddr) {
+        self.connection_address = connection_address.to_string();
+    }
+}
+
+impl Bandwidth {
+    /// Constructs a `BwType` from a string
+    pub fn try_bwtype(&self) -> Result<BwType, ParseEnumError> {
+        BwType::from_str(self.bwtype.as_str())
+    }
+
+    /// Converts a `BwType` to string and sets it to the `bwtype`
+    pub fn set_from_bwtype(&mut self, bwtype: BwType) {
+        self.bwtype = bwtype.to_string();
+    }
+}
+
+impl Key {
+    /// Constructs a `KeyMethod` from a string
+    pub fn try_keymethod(&self) -> Result<KeyMethod, ParseEnumError> {
+        KeyMethod::from_str(self.method.as_str())
+    }
+
+    /// Converts a `KeyMethod` to string and sets it to the `method`
+    pub fn set_from_keymethod(&mut self, method: KeyMethod) {
+        self.method = method.to_string();
+    }
+}
+
+impl Media {
+    /// Constructs a `MediaType` from a string
+    pub fn try_mediatype(&self) -> Result<MediaType, ParseEnumError> {
+        MediaType::from_str(self.media.as_str())
+    }
+
+    /// Converts a `MediaType` to string and sets it to the `media`
+    pub fn set_from_mediatype(&mut self, media: MediaType) {
+        self.media = media.to_string();
+    }
+
+    /// Constructs a `TransportProto` from a string
+    pub fn try_transport_proto(&self) -> Result<TransportProto, ParseEnumError> {
+        TransportProto::from_str(self.proto.as_str())
+    }
+
+    /// Converts a `TransportProto` to string and sets it to the `proto`
+    pub fn set_from_transport_proto(&mut self, proto: TransportProto) {
+        self.proto = proto.to_string();
+    }
+
+    /// Gets an iterator over all attribute values of the given name.
+    /// Each item is a `Result` with the inferred type in `Ok` and `AttributeError` in `Err`
+    /// The payload type can also be passed optionally to filter an Attribute for a particular payload format
+    pub fn get_attribute_values_typed<'a, T: FromStr<Err = AttributeErr>>(
```

Comment:
```suggestion
    pub fn attribute_values_typed<'a, T: FromStr<Err = AttributeErr>>(
```

but maybe also `attributes_typed` or so?